tools/integration_tests/util/client/storage_client.go

```diff
@@ -381,13 +384,26 @@ func ClearCacheControlOnGcsObject(ctx context.Context, client *storage.Client, o
 	return nil
 }
 
+// UploadGcsObject uploads a local file to a specified GCS bucket and object without any preconditions.
+// Handles gzip compression if requested.
+func UploadGcsObject(ctx context.Context, client *storage.Client, localPath, bucketName, objectName string, uploadGzipEncoded bool) error {
+	return UploadGcsObjectWithPreconditions(ctx, client, localPath, bucketName, objectName, uploadGzipEncoded, nil)
+}
+
 func CopyFileInBucket(ctx context.Context, storageClient *storage.Client, srcfilePath, destFilePath, bucket string) {
 	err := UploadGcsObject(ctx, storageClient, srcfilePath, bucket, destFilePath, false)
 	if err != nil {
 		log.Fatalf("Error while copying file %q to GCS object \"gs://%s/%s\" : %v", srcfilePath, bucket, destFilePath, err)
 	}
 }
 
+func CopyFileInBucketWithPreconditions(ctx context.Context, storageClient *storage.Client, srcfilePath, destFilePath, bucket string, preconditions *storage.Conditions) {
+	err := UploadGcsObjectWithPreconditions(ctx, storageClient, srcfilePath, bucket, destFilePath, false, preconditions)
```

Comment:
Why are we calling `UploadGcsObjectWithPreconditions` inside copy function? and if we want to apply `UploadGcsObjectWithPreconditions` function only then can't we directly call that function?